tools/update_glibc.zig

```diff
@@ -0,0 +1,333 @@
+        .path = "s390/s390-64",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{
+            ZigTarget{ .arch = .arm, .abi = .gnueabi },
+            ZigTarget{ .arch = .armeb, .abi = .gnueabi },
+            ZigTarget{ .arch = .arm, .abi = .gnueabihf },
+            ZigTarget{ .arch = .armeb, .abi = .gnueabihf },
+        },
+        .path = "arm",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{
+            ZigTarget{ .arch = .sparc, .abi = .gnu },
+            ZigTarget{ .arch = .sparcel, .abi = .gnu },
+        },
+        .path = "sparc/sparc32",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{ZigTarget{ .arch = .sparcv9, .abi = .gnu }},
+        .path = "sparc/sparc64",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{
+            ZigTarget{ .arch = .mips64el, .abi = .gnuabi64 },
+            ZigTarget{ .arch = .mips64, .abi = .gnuabi64 },
+        },
+        .path = "mips/mips64",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{
+            ZigTarget{ .arch = .mips64el, .abi = .gnuabin32 },
+            ZigTarget{ .arch = .mips64, .abi = .gnuabin32 },
+        },
+        .path = "mips/mips64",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{
+            ZigTarget{ .arch = .mipsel, .abi = .gnueabihf },
+            ZigTarget{ .arch = .mips, .abi = .gnueabihf },
+        },
+        .path = "mips/mips32",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{
+            ZigTarget{ .arch = .mipsel, .abi = .gnueabi },
+            ZigTarget{ .arch = .mips, .abi = .gnueabi },
+        },
+        .path = "mips/mips32",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{ZigTarget{ .arch = .x86_64, .abi = .gnu }},
+        .path = "x86_64/64",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{ZigTarget{ .arch = .x86_64, .abi = .gnux32 }},
+        .path = "x86_64/x32",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{ZigTarget{ .arch = .i386, .abi = .gnu }},
+        .path = "i386",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{ZigTarget{ .arch = .powerpc64le, .abi = .gnu }},
+        .path = "powerpc/powerpc64/le",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{ZigTarget{ .arch = .powerpc64, .abi = .gnu }},
+        .path = "powerpc/powerpc64/be",
+    },
+    AbiList{
+        .targets = [_]ZigTarget{
+            ZigTarget{ .arch = .powerpc, .abi = .gnueabi },
+            ZigTarget{ .arch = .powerpc, .abi = .gnueabihf },
+        },
+        .path = "powerpc/powerpc32",
+    },
+};
+
+const FunctionSet = struct {
+    list: std.ArrayList(VersionedFn),
+    fn_vers_list: FnVersionList,
+};
+const FnVersionList = std.AutoHashMap([]const u8, std.ArrayList(usize));
+
+const VersionedFn = struct {
+    ver: []const u8, // example: "GLIBC_2.15"
+    name: []const u8, // example: "puts"
+};
+const Function = struct {
+    name: []const u8, // example: "puts"
+    lib: []const u8, // example: "c"
```

Comment:
Has it ever happended that a function got moved between libraries? e.g. from libm into libc? Could we handle it if it happened in future?

Reply:
I don't know if it's happened but it's trivially handled by this script. This script parses the .abilist files from glibc source, which describe all symbols. So if a function got moved then the .abilist file in glibc would get updated and this tool would pick up the changes, no problem. This PR actually makes zig *more* resilient to this from master branch.

Reply:
But wouldn't it then be in one library for an old release and in a different library for a new release? What would appears in fns.txt? multiple entries for one function name?

Reply:
I see the question now. Technically some functions are in multiple libraries, and there are some straightforward changes that could be done to respect that. However, when zig links against glibc we always link all the libraries. I'm pretty sure it doesn't matter which library it says a function came from, because at runtime the dynamic linker will find the correct one. If we run into trouble, no problem, I'll do the change to make it support functions in multiple libs.

Reply:
@daurnimator yes, many time functions got moved from librt to libc